eth/handler.go

```diff
@@ -462,35 +463,33 @@ func (h *handler) txBroadcastLoop() {
 	}
 }
 
-// missingParentLoop announces new pendingEtxs to connected peers.
-func (h *handler) missingParentLoop() {
+// missingBlockLoop announces new pendingEtxs to connected peers.
+func (h *handler) missingBlockLoop() {
 	defer h.wg.Done()
 	for {
 		select {
-		case hash := <-h.missingParentCh:
+		case blockRequest := <-h.missingBlockCh:
+			headerRequested := 0
 			// Check if any of the peers have the body
-			for _, peer := range h.selectSomePeers() {
-				log.Trace("Fetching the missing parent from", "peer", peer.ID(), "hash", hash)
-				peer.RequestBlockByHash(hash)
+			allPeers := h.peers.allPeers()
+			// shuffle the filteredPeers
+			rand.Shuffle(len(allPeers), func(i, j int) { allPeers[i], allPeers[j] = allPeers[j], allPeers[i] })
+
+			for _, peer := range allPeers {
+				log.Trace("Fetching the missing parent from", "peer", peer.ID(), "hash", blockRequest.Hash)
+				_, _, peerEntropy, _ := peer.Head()
+				if peerEntropy != nil {
+					if peerEntropy.Cmp(blockRequest.Entropy) > 0 {
+						peer.RequestBlockByHash(blockRequest.Hash)
+						headerRequested++
+					}
+				}
+				if headerRequested == minPeerRequest {
```

Comment:
If the 3 peers you asked don't have the block, do you try again with different peers later?

Reply:
yeah, everytime, random peers